DH-11

Dyrk84 opened the pull request:
New Action menu: running away.
Upgraded nextTurn method.
2 Tests to running away action.

Dyrk84 committed:
DH-11
New Action menu: running away.
Upgraded nextTurn method.
2 Tests to running away action.

gergely-szanto-epam commented on src/main/java/hu/dungeonhunter/tools/Fight.java:
```diff
@@ -31,14 +32,26 @@ public Fight(int numOfMonsters) {  //for test
 
-    private void monsterIncomingOrWin(){
+    public void runningAway() {
+        System.out.println("The monster hits you a last one before you can run away: ");
```
```suggestion
        System.out.println("The monster hits you a last time before you can run away: ");
```

Dyrk84 replied:
I corrected

gergely-szanto-epam commented on src/test/java/hu/dungeonhunter/TestFight.java:
```diff
@@ -36,5 +36,34 @@ public void championDefeatTest() {
+
+        assertThat(fight.getMonsterCounter()).as("monsterCounter test to decrease in value").isEqualTo(0);
+        assertThat(fight.nextTurn()).as("There should be no next turn, because no more monster.").isFalse();
+
+
+
+
+
```
remove these empty lines :)

Dyrk84 replied:
ups :D 
corrected.

gergely-szanto-epam commented on src/test/java/hu/dungeonhunter/TestFight.java:
```diff
@@ -36,5 +36,34 @@ public void championDefeatTest() {
+        fight.runningAway();
+
+        assertThat(fight.getMonsterCounter()).as("monsterCounter test to decrease in value").isEqualTo(0);
+        assertThat(fight.nextTurn()).as("There should be no next turn, because no more monster.").isFalse();
```
Also add assertion to check that champion is not defeated.

Dyrk84 replied:
Thank you, it's done!

Dyrk84 committed:
DH-11
New Action menu: running away.
Upgraded nextTurn method.
2 Tests to running away action.

gergely-szanto-epam commented on src/test/java/hu/dungeonhunter/TestFight.java:
```diff
@@ -37,4 +37,29 @@ public void championDefeatTest() {
+
+        assertThat(fight.getMonsterCounter()).as("monsterCounter test to decrease in value").isEqualTo(0);
+        assertThat(fight.nextTurn()).as("There should be no next turn, because no more monster.").isFalse();
+        assertThat(highHpChampion.isDefeat()).as("Champion should have been  no defeated.").isFalse();
```
```suggestion
        assertThat(highHpChampion.isDefeat()).as("Champion should have been not defeated.").isFalse();
```

Dyrk84 replied:
I corrected.

Dyrk84 committed:
DH-11
New Action menu: running away.
Upgraded nextTurn method.
2 Tests to running away action.
